src/util/manage_nsfs_events_utils.js

```diff
@@ -0,0 +1,266 @@
+            severity: 'ERROR',
+            state: 'DEGRADED',
+        },
+        BUCKET_DIR_CREATION_FAILED: {
+            event_code: 'noobaa_bucket_dir_creation_failed',
+            entity_type: 'NODE',
+            event_type: 'ERROR',
+            message: 'Could not create underlying bucket directory.',
+            description: 'Could not create underlying bucket directory, Check for permission and dir path,',
+            scope: 'NODE',
+            severity: 'ERROR',
+            state: 'DEGRADED',
+        },
+        BUCKET_DELETE_FAILED: {
+            event_code: 'noobaa_bucket_delete_failed',
+            entity_type: 'NODE',
+            event_type: 'ERROR',
+            message: 'Could not delete underlying bucket.',
+            description: 'Could not create underlying bucket.',
+            scope: 'NODE',
+            severity: 'ERROR',
+            state: 'DEGRADED',
+        },
+        NO_SUCH_BUCKET: {
+            event_code: 'noobaa_no_such_bucket',
+            message: 'Bucket not found',
+            description: 'Bucket not found',
+            entity_type: 'NODE',
+            event_type: 'ERROR',
+            scope: 'NODE',
+            severity: 'ERROR',
+            state: 'HEALTHY',
+        },
+        BUCKET_ALREADY_EXISTS: {
+            event_code: 'noobaa_bucket_already_exists',
+            message: 'Bucket already exists with the name.',
+            description: 'Bucket already exists with the name.',
+            entity_type: 'NODE',
+            event_type: 'ERROR',
+            scope: 'NODE',
+            severity: 'ERROR',
+            state: 'HEALTHY',
+        },
+        UNAUTHORIZED: {
+            event_code: 'noobaa_bucket_access_unauthorized',
+            message: 'Bucket is not accessible with current access rights. ',
+            description: 'Bucket is not accessible with current access rights.',
+            entity_type: 'NODE',
+            event_type: 'ERROR',
+            scope: 'NODE',
+            severity: 'ERROR',
+            state: 'HEALTHY',
+        },
+        IO_STREAM_ITEM_TIMEOUT: {
+            event_code: 'bucket_io_stream_item_timeout',
+            message: 'Bucket IO sream timeout.',
+            description: 'Bucket IO sream timeout.',
+            entity_type: 'NODE',
+            event_type: 'ERROR',
+            scope: 'NODE',
+            severity: 'ERROR',
+            state: 'HEALTHY',
+        },
+        INTERNAL_ERROR: {
+            event_code: 'bucket_internal_error',
+            message: 'Bucket action failed with internal error.',
+            description: 'Bucket IO sream timeout.',
+            entity_type: 'NODE',
+            event_type: 'ERROR',
+            scope: 'NODE',
+            severity: 'ERROR',
+            state: 'HEALTHY',
+        }
+    }
+};
+
+async function create_event(event_from, event_code, name, arg, err) {
+    const description = EVENT_ERROR_CODE[event_from][event_code].description + (err ? ', error: ' + err : '');
```

Comment:
worth adding a variable, so we won't need to evaluate this every time -
const event = EVENT_ERROR_CODE[event_from][event_code];
and then use event.event_code/ event.entity_type / etc...